syft/pkg/cataloger/generic/cataloger.go

```diff
@@ -167,11 +166,12 @@ func (c *Cataloger) Catalog(ctx context.Context, resolver file.Resolver) ([]pkg.
 		pkgs []pkg.Package
 		rels []artifact.Relationship
 	}
-	errs = sync.Collect(&ctx, cataloging.ExecutorFile, sync.ToSeq(c.selectFiles(resolver)), func(req request) (result, error) {
+	errs := sync.Collect(&ctx, cataloging.ExecutorFile, sync.ToSeq(c.selectFiles(resolver)), func(req request) (result, error) {
 		location, parser := req.Location, req.Parser
 
 		log.WithFields("path", location.RealPath).Trace("parsing file contents")
 
+		var errs error
```

Comment:
instead of this, we could just change the line below to:

```
err = unknown.New(location, err)
```

Reply:
I can do that as well 👍

Reply:
actually, I shouldn't do that... I'd be appending the error onto what I'm writing to

Reply:
It's `unknown.New` rather than `Append`